src/tools/sfdk/sdkmanager.cpp

```diff
@@ -310,6 +314,87 @@ void SdkManager::setEnableReversePathMapping(bool enable)
     s_instance->m_enableReversePathMapping = enable;
 }
 
+Device *SdkManager::configuredDevice(QString *errorMessage)
+{
+    Q_ASSERT(errorMessage);
+
+    const Option *const deviceOption = Dispatcher::option("device"); // FIXME
+    QTC_ASSERT(deviceOption, return nullptr);
+
+    const Utils::optional<OptionEffectiveOccurence> effectiveDeviceOption =
+        Configuration::effectiveState(deviceOption);
+    if (!effectiveDeviceOption) {
+        *errorMessage = tr("No device selected");
+        return nullptr;
+    }
+    QTC_ASSERT(!effectiveDeviceOption->isMasked(), return nullptr);
```

Comment:
Shouldn't we put something into the errorMessage even in this case? It might help debug future bugs.

Reply:
I prefer not doing that - it would be practically a dead code (assertions should only fail on internal errors)